src/client/connection.rs

```diff
@@ -376,22 +375,50 @@ impl<S: AsyncRead + AsyncWrite + Unpin + Send> Connection<S> {
         self,
         config: &Config,
         encryption: EncryptionLevel,
-        trust_cert: bool,
     ) -> crate::Result<Self> {
         if encryption != EncryptionLevel::NotSupported {
             event!(Level::INFO, "Performing a TLS handshake");
 
             let mut builder = TlsConnector::new();
 
-            if trust_cert {
-                event!(
-                    Level::WARN,
-                    "Trusting the server certificate without validation."
-                );
-
-                builder = builder.danger_accept_invalid_certs(true);
-                builder = builder.danger_accept_invalid_hostnames(true);
-                builder = builder.use_sni(false);
+            match &config.trust {
+                TrustConfig::CaCertificateLocation(path) => {
+                    if let Ok(buf) = fs::read(path) {
+                        let cert = match path.extension() {
+                            Some(ext)
+                                if ext.to_ascii_lowercase() == "pem"
+                                    || ext.to_ascii_lowercase() == "crt" =>
+                            {
+                                Some(Certificate::from_pem(&buf)?)
+                            }
+                            Some(ext) if ext.to_ascii_lowercase() == "der" => {
+                                Some(Certificate::from_der(&buf)?)
+                            }
+                            Some(_) | None => {
+                                event!(Level::ERROR, "Unknown CA certificate format. Only 'pem', 'crt' and 'der' supported. Certificate will not be added to trust-chain.");
```

Comment:
Maybe return an error instead, so the user does not have a certificate setup they did not ask for.